nntrainer/models/neuralnet.cpp

```diff
@@ -747,17 +748,37 @@ int NeuralNetwork::extendGraph(GraphType graph, std::string prefix) {
     return ML_ERROR_NOT_SUPPORTED;
   }
 
+  if (graph.size() == 0)
+    return ML_ERROR_NONE;
+
+  for (unsigned int i = 0; i < graph[0]->input_layers.size(); ++i) {
+    if (sub_in_out.find(graph[0]->input_layers[i]) != sub_in_out.end()) {
+      graph[0]->input_layers[i] = sub_in_out[graph[0]->input_layers[i]];
+    }
+  }
+
   /** Insert the layer to the graph */
   for (auto layer : graph) {
     /**
      * Add prefix to the existing layer name,
      * and ensure it is unique in this new graph
      */
+    std::string org_name = prefix + layer->getName();
     ensureName(layer, prefix, true);
+    sub_in_out.insert(std::make_pair(org_name, layer->getName()));
+
+    for (unsigned int i = 0; i < layer->input_layers.size(); ++i) {
+      if (sub_in_out.find(prefix + layer->input_layers[i]) !=
+          sub_in_out.end()) {
+        layer->input_layers[i] = sub_in_out[prefix + layer->input_layers[i]];
```

Comment:
IMO, there seems few points to improve.

1. Renewing input_layers should be done after the loop of creating `sub_in_out` is done. `sub_in_out` might not contain the name it should have (will lead to an error for some cases)
2. For the same reason, not finding the key should throw an error.
3. `sub_in_out` should be a variable inside this function. so that it is cleared when the function starts. And the variable is only used inside this function
4. keys don't need to have the prefix added. It is redundant.

Reply:
1. sub_in_out is not renewed.
for 2, if it is not found then definitely, it is not the error.
about 3, we need it global. using inside this function only is not what I indented. It must be the unique throughout the network.
about 4. due to the reason 3, prefix is required.

Reply:
I wonder if `sub_in_out` has some use outside of this function. Currently, there is no occurrence of `sub_in_out` and there is no other way to access the entry outside of this function (since key is processed inside this function and not saved elsewhere).

Plus, there are some issues when this stays global.

1. we need to ensure `org_name` is unique which seems not guaranteed, line 767 gaurantee's it is unique, so we cannot be sure if it is unique before line 767.

and for 🌵, there will be a hidden error if input layer is processed after the current layer, which is hard to find the cause of the problem.
e.g)
```ini

[a]
input_layers=b

[b]
...
```
ini sections does not have guaranteed order so, even if [b] is before [a], it might be processed before [a]. So I think not finding new names for the input layers should throw

So updating `sub_in_out` should be done like

```cpp
for (auto& layer : graph) {
 update sub_in_out
}

for(auto& layer: graph) {
 update layer information
}
```

Reply:
I think ini does not guarantee problem is not just for this. we have to handle this sepeartely. sub_in_out need be global. we have to consider that used same sub_graph repeatly with different section name.

Reply:
Understood the reason behind the code. Thanks